notebooks/DM-45573_Depth_Estimates.ipynb

```diff
@@ -0,0 +1,798 @@
+    "2. The calibrations are (slightly) different between the SourceCatalog (`base_LocalPhotoCalib`) and the SourceTable (`localPhotoCalib`). This may not be surprising if the calibrations are run multiple times.\n",
+    "3. It is possible to recover the `psfFlux` using the `localPhotoCalib`. This can be formally or informally. See below for an example."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "id": "9ca5cbc3-303f-486b-8f2d-a5411527b677",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2024-08-27T21:42:58.877841Z",
+     "iopub.status.busy": "2024-08-27T21:42:58.877531Z",
+     "iopub.status.idle": "2024-08-27T21:42:58.882032Z",
+     "shell.execute_reply": "2024-08-27T21:42:58.881662Z",
+     "shell.execute_reply.started": "2024-08-27T21:42:58.877823Z"
+    }
+   },
+   "source": [
+    "from lsst.pipe.tasks.functors import LocalPhotometry\n",
+    "phot = LocalPhotometry(table['slot_PsfFlux_instFlux'], srcTable['localPhotoCalib'],\n",
+    "                       table['base_LocalPhotoCalib'], srcTable['localPhotoCalib'])\n",
+    "psfFlux = phot.instFluxToNanojansky(table['slot_PsfFlux_instFlux'], srcTable['localPhotoCalib'])\n",
+    "assert np.allclose(srcTable['psfFlux'], psfFlux, equal_nan=True)\n",
+    "\n",
+    "psfFlux = table['slot_PsfFlux_instFlux'] * srcTable['localPhotoCalib']\n",
+    "assert np.allclose(srcTable['psfFlux'], psfFlux, equal_nan=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e11a4140-ee51-4a50-a67a-e4663664e050",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert flux to magnitude\n",
+    "flux = srcTable['psfFlux']\n",
+    "fluxerr = srcTable['psfFluxErr']\n",
+    "\n",
+    "# Convert from nJy to AB mag\n",
+    "with np.errstate(invalid='ignore'):\n",
+    "    nJytoAB = (1*u.nJy).to(u.ABmag).value\n",
+    "    mag = -2.5*np.log10(flux) + nJytoAB\n",
+    "    magerr = 2.5/np.log(10) * fluxerr / flux\n",
+    "\n",
+    "snr = flux/fluxerr\n",
+    "srcTable['psfMag'] = mag\n",
+    "srcTable['psfMagErr'] = magerr\n",
+    "srcTable['psfFluxSnr'] = snr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "004f89e8-3d90-49c4-92cc-3bd30fe88f32",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select stars using extendedness (follows analysis_tools)\n",
+    "#extendedness = table['base_ClassificationExtendedness_value']\n",
+    "extendedness = srcTable['extendedness']\n",
+    "star = (extendedness >= 0) & (extendedness < 0.5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "342086d7-cebf-44ff-8746-4d146a5230ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Perform the catalog determination\n",
+    "snrval = 5.0\n",
+    "snrmin, snrmax = 4.75, 5.25\n",
```

Comment:
I notice the bin for selecting source to use in the catalog determination is pretty large, 0.5 magnitudes, and that it's also selected from only one detector in one visit. I'd be interested to see whether using a smaller bin and getting more statistics (if needed) from additional detectors or visits would change anything. (Too large a bin could change the results, because the distribution of sources across the bin will not be uniform.)

Reply:
The bin size is 0.5 in *SNR*. Sources passing this selection span 0.09 mag. The bin size was chosen to match the [FiveSigmaPointSourceDepthMetric](https://github.com/lsst/analysis_tools/blob/23501cb5eabd911f92de508725b66d903a26672e/python/lsst/analysis/tools/atools/limitingMagnitudeMetric.py#L31) in analysis_tools, but there is no strong motivation. You are certainly right that a smaller bin would give a more precise result, but I don't really trust the magnitude limit estimates to better than 0.1 mag anyway.